source/NuLink.Cli/LinkCommand.cs

```
@@ -15,47 +16,86 @@ public LinkCommand(IUserInterface ui)
+                    LinkPackage(package, localProjectPath, false);
+                }
+            }
+        }
+
+        private void LinkPackage(PackageReferenceInfo requestedPackage, string localProjectPath, bool singleMode)
+        {
+            if (localProjectPath == null)
+            {
+                _ui.Report(singleMode ? VerbosityLevel.Error : VerbosityLevel.Low, () =>
+                    $"{(singleMode ? string.Concat(VerbosityLevel.Error.ToString(), ": ") : string.Empty)}" +
+                    $"Cannot find corresponding project to package {requestedPackage.PackageId}");
+
+                return;
+            }
+
+            var linkTargetPath = Path.Combine(Path.GetDirectoryName(localProjectPath), "bin", "Debug");
             var status = requestedPackage.CheckStatus();
-            var linkTargetPath = Path.Combine(Path.GetDirectoryName(options.LocalProjectPath), "bin", "Debug");
 
-            ValidateOperation();
+            if (!ValidateOperation())
+            {
+                return;
+            }
+
             PerformOperation();
-            
+
             _ui.ReportSuccess(() => $"Linked {requestedPackage.LibFolderPath}");
             _ui.ReportSuccess(() => $" -> {linkTargetPath}", ConsoleColor.Magenta);
-            return 0;
 
-            PackageReferenceInfo GetPackageInfo()
-            {
-                var allProjects = new WorkspaceLoader().LoadProjects(options.ConsumerProjectPath, options.ProjectIsSolution);
-                var referenceLoader = new PackageReferenceLoader(_ui);
-                var allPackages = referenceLoader.LoadPackageReferences(allProjects);
-                var package = allPackages.FirstOrDefault(p => p.PackageId == options.PackageId);
-                return package ?? throw new Exception($"Error: Package not referenced: {options.PackageId}");
-            }
-            
-            void ValidateOperation()
+            bool ValidateOperation()
```

TL;DR: let's keep throwing exceptions rather than returning a `bool`.

I was thinking quite a bit about this one. If we want to support both "ignore errors" and "fail fast" modes, the code can become complicated. I mean that using exception results in simpler code. On the other hand, we have "do not use exceptions to steer logic". 

In the bottom line, I think that throwing exceptions from `LinkPackage()` is OK, because `LinkPackage` should either perform its responsibility or throw. We can implement both "fail fast" and "ignore" by catching at the caller (inside the `foreach` loop). When in "ignore" mode, report a warning. When in "fail fast" mode, re-throw.